src/extensions/extension_manager/src/extension_manager_modules.cpp

```diff
@@ -0,0 +1,237 @@
+/**
+ * @file extension_manager_modules.cpp
+ * @brief Implementation of ExtensionManager.
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+#include "aduc/extension_manager.h"
+
+#include <aduc/c_utils.h>
+#include <aduc/calloc_wrapper.hpp> // ADUC::StringUtils::cstr_wrapper
+#include <aduc/component_enumerator_extension.hpp>
+#include <aduc/config_utils.h>
+#include <aduc/content_downloader_extension.hpp>
+#include <aduc/content_handler.hpp>
+#include <aduc/contract_utils.h>
+#include <aduc/exceptions.hpp>
+#include <aduc/exports/extension_export_symbols.h>
+#include <aduc/extension_manager.hpp>
+#include <aduc/extension_manager_helper.hpp>
+#include <aduc/extension_utils.h>
+#include <aduc/hash_utils.h> // for SHAversion
+#include <aduc/logging.h>
+#include <aduc/parser_utils.h>
+#include <aduc/path_utils.h> // SanitizePathSegment
+#include <aduc/plugin_exception.hpp>
+#include <aduc/result.h>
+#include <aduc/string_c_utils.h>
+#include <aduc/string_handle_wrapper.hpp>
+#include <aduc/string_utils.hpp>
+#include <aduc/types/workflow.h> // ADUC_WorkflowHandle
+#include <aduc/workflow_utils.h>
+
+#include <cstring>
+#include <unordered_map>
+
+// Note: this requires ${CMAKE_DL_LIBS}
+#include <dlfcn.h>
+#include <unistd.h>
+
+#include <dirent.h> // for opendir, readdir, closedir, DIR
```

Comment:
must be `#include <aducpal/dirent.h>` to be cross-platform

Reply:
This file is removed.